jooqdb/src/main/java/org/cristalise/storage/jooqdb/JooqClusterStorage.java

```diff
@@ -1,339 +1,426 @@
+        }
+
+        if (! Gateway.getProperties().getBoolean(JOOQ_DISABLE_DOMAIN_CREATE, false)) {
+            context.transaction(nested -> {
+                for (JooqDomainHandler handler: domainHandlers) handler.createTables(DSL.using(nested));
+            });
+        }
+    }
+
+    public void dropHandlers() throws PersistencyException {
+        DSLContext context = JooqHandler.connect();
+        for (JooqHandler handler: jooqHandlers.values()) handler.dropTables(context);
+    }
+
+    @Override
+    public void close() throws PersistencyException {
+        Logger.msg(1, "JooqClusterStorage.close()");
+        try {
+            JooqHandler.connect().close();
+        }
+        catch (Exception e) {
+            Logger.error(e);
+            throw new PersistencyException(e.getMessage());
+        }
+    }
+
+    @Override
+    public void postBoostrap() throws PersistencyException {
+        JooqHandler.connect().transaction(nested ->{
+            for (JooqDomainHandler domainHandler : domainHandlers) domainHandler.postBoostrap(DSL.using(nested));
+        });
+        
+        if(Gateway.getProperties().containsKey(JOOQ_AUTOCOMMIT)){
+
+            Gateway.getProperties().put(JOOQ_AUTOCOMMIT, false);
+            HikariConfig config = new HikariConfig();
+            JooqHandler.ds.copyStateTo(config);
+            config.setAutoCommit(false);
+            config.addDataSourceProperty( "autoCommit",false);
+            HikariDataSource newDs = new HikariDataSource(config);    
+            JooqHandler.ds = newDs;
+           
+        }
+    }
+
+    @Override
+    public void postStartServer() throws PersistencyException {
+        JooqHandler.connect().transaction(nested ->{
+            for (JooqDomainHandler domainHandler : domainHandlers) domainHandler.postStartServer(DSL.using(nested));
+        });
+    }
+
+    @Override
+    public void postConnect() throws PersistencyException {
+        if(Gateway.getProperties().containsKey(JOOQ_AUTOCOMMIT)){
+            Gateway.getProperties().put(JOOQ_AUTOCOMMIT, true);
+        }
+        JooqHandler.connect().transaction(nested ->{
+            for (JooqDomainHandler domainHandler : domainHandlers) domainHandler.postConnect(DSL.using(nested));
+        });
+        
+    }
+
+    @Override
+    public void begin(Object locker) {
+        try {
+            DSLContext context = JooqHandler.connect();
+            contextMap.put(locker, context);
+
+            if (Logger.doLog(5)) JooqHandler.logConnectionCount("JooqClusterStorage.begin()", context);
+        }
+        catch (PersistencyException e) {
+            Logger.error(e);
+        }
+    }
+
+    @Override
+    public void commit(Object locker) throws PersistencyException {
+        DSLContext context = null;
+        
+        autoCommit =  Gateway.getProperties().getBoolean(JOOQ_AUTOCOMMIT);
+        if(!Objects.isNull(locker) && !contextMap.isEmpty()){

```

Comment:
I don't think there's any need for this  `if`. If locker is null or the map is empty, the `get` will return `null`, and the `?:` operator already handles that case.

Reply:
Changes will be on next commit.